File: week00/day01/practice.cpp
#include <array>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int n;
    cin >> n;

    vector<long long> numbers(n);

    for (long long& number : numbers) {
        cin >> number;
    }

    long long total = 0;
    long long min_value = numbers[0];
    long long max_value = numbers[0];

    for (long long number : numbers) {
        total += number;

        if (number < min_value) {
            min_value = number;
        }

        if (number > max_value) {
            max_value = number;
        }
    }

    string text;
    cin >> text;

    array<int, 26> frequency{};

    for (char ch : text) {
        int index = ch - 'a';
        frequency[index]++;
    }

    cout << total << ' '
         << min_value << ' '
         << max_value << '\n';

    for (int i = 0; i < 26; i++) {
        if (frequency[i] > 0) {
            char ch = static_cast<char>(i + 'a');
            cout << ch << ' ' << frequency[i] << '\n';
        }
    }

    return 0;
        
}

File: week00/day01/practice.py
import sys

input = sys.stdin.readline

def solve1():
    n = int(input())
    numbers = list(map(int, input().split()))

    total = 0
    min_value = numbers[0]
    max_value = numbers[0]

    for number in numbers:
        total += number

        if number < min_value:
            min_value = number

        if number > max_value:
            max_value = number

    text = input().strip()

    frequency = [0] * 26

    for ch in text:
        index = ord(ch) - ord('a')
        frequency[index] += 1

    print(total, min_value, max_value)

    for i in range(26):
        if frequency[i] > 0:
            ch = chr(ord('a') + i)
            print(ch, frequency[i])

def solve2():
    n = int(input())
    numbers = list(map(int, input().split()))
    text = input().strip()

    total = sum(numbers)
    min_value = min(numbers)
    max_value = max(numbers)

    frequency = [0] * 26

    for ch in text:
        frequency[ord(ch) - ord('a')] += 1

    print(total, min_value, max_value)

    for i, count in enumerate(frequency):
        if count > 0:
            print(chr(ord('a') + i), count)

if __name__ == "__main__":
    solve1()